nym-vpn-lib/src/credentials/helpers.rs

```diff
@@ -0,0 +1,59 @@
+use std::path::PathBuf;
+
+use nym_credential_storage::persistent_storage::PersistentStorage;
+
+use nym_sdk::{mixnet::StoragePaths, NymNetworkDetails};
+use nym_validator_client::{
+    coconut::{all_coconut_api_clients, CoconutApiError},
+    nyxd::{error::NyxdError, Config as NyxdClientConfig, NyxdClient},
+    QueryHttpRpcNyxdClient,
+};
+use tracing::debug;
+
+use super::CredentialError;
+
+pub(super) async fn get_credentials_store(
+    data_path: PathBuf,
+) -> Result<(PersistentStorage, PathBuf), CredentialError> {
+    let storage_path = StoragePaths::new_from_dir(data_path)?;
+    let credential_db_path = storage_path.credential_database_path;
+    debug!("Credential store: {}", credential_db_path.display());
+    Ok((
+        nym_credential_storage::initialise_persistent_storage(credential_db_path.clone()).await,
+        credential_db_path,
+    ))
+}
+
+pub(super) fn get_nyxd_client() -> Result<QueryHttpRpcNyxdClient, CredentialError> {
+    let network = NymNetworkDetails::new_from_env();
```

Comment:
just to make sure: in mobile you're setting the env to have this info available, right?